extensions/README.md

```diff
@@ -0,0 +1,391 @@
+}
+
+var sampleText = `
+{
+      "name": "31fb5abf-023e-4992-aa4e-95f9c1503bfa",
+      "regid": "https://acme.example",
+      "email": "info@acme.com",
+      "roles": [
+        "tagCreator",
+        "creator",
+        "maintainer"
+      ]
+}
+`
+
+func main() {
+	var entity comid.Entity
+	entity.RegisterExtensions(&EntityExtensions{})
+
+	if err := json.Unmarshal([]byte(sampleText), &entity); err != nil {
+		log.Fatalf("ERROR: %s", err.Error())
+	}
+
+	if err := entity.Valid(); err != nil {
+		log.Fatalf("failed to validate: %s", err.Error())
+	} else {
+		fmt.Println("validation succeeded")
+	}
+
+    // obtain the extension field value via a generic getter
+	email := entity.Extensions.MustGetString("email")
+	fmt.Printf("entity email: %s\n", email)
+
+    // retrive the extensions struct and get value via its field.
+	exts := entity.GetExtensions().(*EntityExtensions)
+	fmt.Printf("also entity email: %s\n", exts.Email)
+}
+```
+
+
+## Type Choice Extensions
+
+Type Choice extensions allow specifying alternative types for existing CoRIM
+fields by defining a type that implements an appropriate interface and
+registering it with CBOR tag.
+
+A type choice struct contains a single field, `Value`, that contains the actual
+object represented by the type choice. The `Value` implements an interface
+that is specific to the type choice and is derived from `ITypeChoiceValue`:
+
+```go
+type ITypeChoiceValue interface {
+	// String returns the string representation of the ITypeChoiceValue.
+	String() string
+	// Valid returns an error if validation of the ITypeChoiceValue fails,
+	// or nil if it succeeds.
+	Valid() error
+	// Type returns the type name of this ITypeChoiceValue implementation.
+	Type() string
+}
+```
+
+The following is the full list of type choice structs:
+
+- `comid.ClassID`
+- `comid.CryptoKey`
+- `comid.EntityName`
+- `comid.Group`
+- `comid.Instance`
+- `comid.Mkey`
+- `comid.SVN`
+- `corim.EntityName`
+
+To provide a new value type, the following is required:
+
+1. Define a type that implements the value interface for the type choice you
+   want to extend. This interface is called `I<NAME>Value`,  where `<NAME>` is
+   the name of the type choice type(e.g. `IClassIDValue`).  These interfaces
+   always embed `ITypeChoiceValue` and possible define additional methods.
+2. Create a factory function for your type, with the signature `func (any)
+   (*<NAME>, error)`, where `<NAME>` is the name of the type choice type that
+   will contain your value. (Note that the function must return a pointer to
+   the container type choice struct, _not_ to the value type you define.) This
+   function should create an instance of your value type from the provided
+   input and return a new type choice struct instance containing it. The range
+   of valid inputs is up to you, however it _must_ handle `null`, returning the
+   [zero-value](https://go.dev/ref/spec#The_zero_value) creating the zero value
+   for your type in that case.
```

Comment:
_"[...] however it _must_ handle `null`, returning the [zero-value](https://go.dev/ref/spec#The_zero_value) creating the zero value for your type in that case."_

There seems to be some missing punctuation, or some amount of redundancy, which makes it difficult to understand.

Reply:
urgh, yeah, the "creating the zero value" bit is redundant.